buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -372,18 +372,20 @@ class BuildPlugin implements Plugin<Project> {
                 options.fork = true
                 options.forkOptions.executable = new File(project.javaHome, 'bin/javac')
                 options.forkOptions.memoryMaximumSize = "1g"
+                if (targetCompatibility == '1.8' || targetCompatibility == '1.9') {
```

Comment:
We actually set targetCompatibility to a value from gradle's JavaVersion enum. So this comparison can be something like:
```
if (targetCompatibility >= JavaVersion.VERSION_1_8) {
```